### PR TITLE
Compute update with 2024-03-02 DiskRP and 2024-07-01 ComputeRP updates

### PR DESCRIPTION
**Pull request by grizzlytheodore**
This is updating SDK with two new versions in compute: 
DiskRP 2024-03-02 version https://github.com/Azure/azure-rest-api-specs/pull/29399
ComputeRP 2024-07-01 version https://github.com/Azure/azure-rest-api-specs/pull/29396

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by grizzlytheodore** [comment]
@archerzz  @ArcturusZhang   please review this PR !

**Comment by grizzlytheodore** [comment]
> @archerzz @ArcturusZhang I have CI failure for not having beta version in changelog, we dont need a beta release, what can i do?

please help me resolve the CI error on changelog

File: sdk/compute/Azure.ResourceManager.Compute/src/ApiCompatBaseline.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# The attribute is in shared source which is included and compiled into the binary, so there is no compatibility concern on it.
+MembersMustExist : Member 'public void Azure.ResourceManager.Compute.Models.ComputeSecurityPostureReference..ctor()' does not exist in the implementation but it does exist in the contract.
+MembersMustExist : Member 'public System.Collections.Generic.IList<Azure.ResourceManager.Compute.VirtualMachineExtensionData> Azure.ResourceManager.Compute.Models.ComputeSecurityPostureReference.ExcludeExtensions.get()' does not exist in the implementation but it does exist in the contract.
```
Comment (ArcturusZhang) [inline sdk/compute/Azure.ResourceManager.Compute/src/ApiCompatBaseline.txt:3]: please remove this file, we should not resolve the breaking changes in this way

**Comment by grizzlytheodore** [comment]
@ArcturusZhang  could you please review this PR?

**Comment by grizzlytheodore** [comment]
> @archerzz @ArcturusZhang could you please review this PR?

pls review/merge !

File: sdk/compute/Azure.ResourceManager.Compute/api/Azure.ResourceManager.Compute.netstandard2.0.cs [hunk 2]
```diff
@@ -7357,6 +7406,27 @@ public SharingUpdate(Azure.ResourceManager.Compute.Models.SharingUpdateOperation
         public static bool operator !=(Azure.ResourceManager.Compute.Models.SharingUpdateOperationType left, Azure.ResourceManager.Compute.Models.SharingUpdateOperationType right) { throw null; }
         public override string ToString() { throw null; }
     }
+    public partial class SkuProfile : System.ClientModel.Primitives.IJsonModel<Azure.ResourceManager.Compute.Models.SkuProfile>, System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.Compute.Models.SkuProfile>
```
Comment (ArcturusZhang) [inline sdk/compute/Azure.ResourceManager.Compute/api/Azure.ResourceManager.Compute.netstandard2.0.cs:7409]: to be consistent, could we rename this to `ComputeSkuProfile`?

File: sdk/compute/Azure.ResourceManager.Compute/api/Azure.ResourceManager.Compute.netstandard2.0.cs [hunk 3]
```diff
@@ -7357,6 +7406,27 @@ public SharingUpdate(Azure.ResourceManager.Compute.Models.SharingUpdateOperation
...
+        void System.ClientModel.Primitives.IJsonModel<Azure.ResourceManager.Compute.Models.SkuProfile>.Write(System.Text.Json.Utf8JsonWriter writer, System.ClientModel.Primitives.ModelReaderWriterOptions options) { }
+        Azure.ResourceManager.Compute.Models.SkuProfile System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.Compute.Models.SkuProfile>.Create(System.BinaryData data, System.ClientModel.Primitives.ModelReaderWriterOptions options) { throw null; }
+        string System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.Compute.Models.SkuProfile>.GetFormatFromOptions(System.ClientModel.Primitives.ModelReaderWriterOptions options) { throw null; }
+        System.BinaryData System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.Compute.Models.SkuProfile>.Write(System.ClientModel.Primitives.ModelReaderWriterOptions options) { throw null; }
+    }
+    public partial class SkuProfileVmSize : System.ClientModel.Primitives.IJsonModel<Azure.ResourceManager.Compute.Models.SkuProfileVmSize>, System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.Compute.Models.SkuProfileVmSize>
```
Comment (ArcturusZhang) [inline sdk/compute/Azure.ResourceManager.Compute/api/Azure.ResourceManager.Compute.netstandard2.0.cs:7420]: Here as well `ComputeSkuProfileVmSize`

Comment (grizzlytheodore) [reply sdk/compute/Azure.ResourceManager.Compute/api/Azure.ResourceManager.Compute.netstandard2.0.cs:7420]: renamed !

File: sdk/compute/Azure.ResourceManager.Compute/api/Azure.ResourceManager.Compute.netstandard2.0.cs (hunk 2 above)
Comment (grizzlytheodore) [reply sdk/compute/Azure.ResourceManager.Compute/api/Azure.ResourceManager.Compute.netstandard2.0.cs]: renamed

File: sdk/compute/Azure.ResourceManager.Compute/CHANGELOG.md [hunk 4]
```diff
@@ -1,14 +1,20 @@
...
+- Added a new property `ScheduledEventsPolicy` to `AvailabilitySetPatch` and `AvailabilitySetData` class.
+
 ### Breaking Changes
 
-### Bugs Fixed
+- `ExcludeExtensions` property in `ComputeSecurityPostureReference` is replaced by `ExcludeExtensionNames` property which takes in a list of strings.
+- `IsVmAgentPlatformUpdatesEnabled` property in `WindowsConfiguration` class is now read-only.
 
-### Other Changes
+## 1.6.0-beta.1 (Not Released)
```
Comment (ArcturusZhang) [inline sdk/compute/Azure.ResourceManager.Compute/CHANGELOG.md:17]: we should remove this line.

File: sdk/compute/Azure.ResourceManager.Compute/src/autorest.md [hunk 5]
```diff
@@ -278,6 +278,8 @@ rename-mapping:
   GalleryArtifactVersionFullSource.virtualMachineId: -|arm-id
   SecurityPostureReference: ComputeSecurityPostureReference
   SecurityPostureReference.excludeExtensions: ExcludeExtensionNames
+  VirtualMachineScaleSet.properties.skuProfile : ComputeSkuProfile
```
Comment (ArcturusZhang) [inline sdk/compute/Azure.ResourceManager.Compute/src/autorest.md:281]: my comment is not asking to change the name of the property. We should change the name of the type of the property.
Sorry to be ambiguous, could you take a look again?
I added another comment to that place, and please revert this

File: sdk/compute/Azure.ResourceManager.Compute/api/Azure.ResourceManager.Compute.netstandard2.0.cs [hunk 6]
```diff
@@ -7357,6 +7416,17 @@ public SharingUpdate(Azure.ResourceManager.Compute.Models.SharingUpdateOperation
         public static bool operator !=(Azure.ResourceManager.Compute.Models.SharingUpdateOperationType left, Azure.ResourceManager.Compute.Models.SharingUpdateOperationType right) { throw null; }
         public override string ToString() { throw null; }
     }
+    public partial class SkuProfile : System.ClientModel.Primitives.IJsonModel<Azure.ResourceManager.Compute.Models.SkuProfile>, System.ClientModel.Primitives.IPersistableModel<Azure.ResourceManager.Compute.Models.SkuProfile>
```
Comment (ArcturusZhang) [inline sdk/compute/Azure.ResourceManager.Compute/api/Azure.ResourceManager.Compute.netstandard2.0.cs:7419]: so this type should be `ComputeSkuProfile`

File: sdk/compute/Azure.ResourceManager.Compute/api/Azure.ResourceManager.Compute.netstandard2.0.cs [hunk 7]
```diff
@@ -2441,6 +2443,7 @@ public partial class VirtualMachineScaleSetData : Azure.ResourceManager.Models.T
         public VirtualMachineScaleSetData(Azure.Core.AzureLocation location) { }
         public Azure.ResourceManager.Compute.Models.AdditionalCapabilities AdditionalCapabilities { get { throw null; } set { } }
         public Azure.ResourceManager.Compute.Models.AutomaticRepairsPolicy AutomaticRepairsPolicy { get { throw null; } set { } }
+        public Azure.ResourceManager.Compute.Models.SkuProfile ComputeSkuProfile { get { throw null; } set { } }
```
Comment (ArcturusZhang) [inline sdk/compute/Azure.ResourceManager.Compute/api/Azure.ResourceManager.Compute.netstandard2.0.cs:2446]: we need to change this properyt name back

File: sdk/compute/Azure.ResourceManager.Compute/api/Azure.ResourceManager.Compute.netstandard2.0.cs [hunk 8]
```diff
@@ -3160,6 +3164,24 @@ public AdditionalUnattendContent() { }
         public override string ToString() { throw null; }
     }
     [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct AllocationStrategy : System.IEquatable<Azure.ResourceManager.Compute.Models.AllocationStrategy>
```
Comment (ArcturusZhang) [inline sdk/compute/Azure.ResourceManager.Compute/api/Azure.ResourceManager.Compute.netstandard2.0.cs:3167]: could we find a better name for this enum as well?

File: sdk/compute/Azure.ResourceManager.Compute/CHANGELOG.md (hunk 4 above)
Comment (grizzlytheodore) [reply sdk/compute/Azure.ResourceManager.Compute/CHANGELOG.md]: i removed it but CI is failing now. which is why I added this line in the first place. 
Could you update it the way it needs to? 

We have 8/23 as release date that we want to meet

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**